libvisual/libvisual/lv_audio.c

```diff
@@ -1336,12 +1344,6 @@ static int input_interleaved_stereo (VisAudioSamplePool *samplepool, VisBuffer *
 	else
 		return -1;
 
-	visual_log_return_val_if_fail (chan1 != NULL, -1);
-	visual_log_return_val_if_fail (chan2 != NULL, -1);
-
-	visual_buffer_set_destroyer (chan1, visual_buffer_destroyer_free);
-	visual_buffer_set_destroyer (chan2, visual_buffer_destroyer_free);
```

Comment:
why are these hooks removed? the commit message does not say.

Reply:
They were duplicate, the `STEREO_INTERLEAVED` macro body already sets the destroyer to `visual_buffer_destroyer_free` in e.g. `chan1 = visual_buffer_new_allocate (bytes_needed_per_channel, visual_buffer_destroyer_free);` the last parameter. Good question though, I will extend the commit message accordingly :+1: